src/pages/Hacker/Checkin.tsx

```diff
@@ -11,89 +11,21 @@ import { generateHackPass } from '../../util';
 import { Email } from '../../features/Checkin/Email';
 import { Reader } from '../../features/Checkin/Reader';
 
-interface ICheckinState {
-  loading: boolean;
-  lastScan: string;
-}
 
-class CheckinContainer extends React.Component<{}, ICheckinState> {
-  constructor(props: {}) {
-    super(props);
-    this.state = {
-      loading: false,
-      lastScan: '',
-    };
-    this.handleScanError = this.handleScanError.bind(this);
-    this.handleScan = this.handleScan.bind(this);
-    this.handleEmail = this.handleEmail.bind(this);
-    this.checkinHacker = this.checkinHacker.bind(this);
-  }
-  public render() {
-    return (
-      <Flex flexDirection={'column'}>
-        <Box>
-          <H1
-            fontSize={'30px'}
-            textAlign={'center'}
-            marginBottom={'20px'}
-            marginLeft={'0px'}
-          >
-            Check in Hacker
-          </H1>
-        </Box>
-        <Box>
-          <Flex
-            flexWrap={'wrap'}
-            justifyContent={'space-evenly'}
-            alignItems={'flex-start'}
-          >
-            <MaxWidthBox maxWidth={'330px'} width={1}>
-              <H1
-                color={theme.colors.black80}
-                fontSize={'24px'}
-                textAlign={'left'}
-                marginBottom={'20px'}
-                marginLeft={'0px'}
-              >
-                By QR Code:
-              </H1>
-              <Box>
-                <Reader
-                  onError={this.handleScanError}
-                  onScan={this.handleScan}
-                />
-              </Box>
-            </MaxWidthBox>
-            <MaxWidthBox maxWidth={'330px'} width={1}>
-              <H1
-                color={theme.colors.black80}
-                fontSize={'24px'}
-                textAlign={'left'}
-                marginBottom={'20px'}
-                marginLeft={'0px'}
-              >
-                By Email:
-              </H1>
-              <Box alignSelf={'center'}>
-                <Email onSubmit={this.handleEmail} />
-              </Box>
-            </MaxWidthBox>
-          </Flex>
-        </Box>
-      </Flex>
-    );
-  }
+const CheckinPage: React.FC = () => {
+  const [loading, setLoading] = useState<boolean>(false);
```

Comment:
consistency issue, do we want to keep it as loading or isLoading?

Reply:
I personally prefer `loading`, but wouldn't mind `isLoading`
We should decide now tho and keep it consistent
@loreina @pengyk @ashaduzzamane @pengyk thoughts?

Reply:
I think `isLoading` like other boolean states (i.e. `isSubmitted`) usually follow the syntax for `is<state>`. if we're using `is` for other states, we should probably keep it even tho I agree that `loading` is cleaner unless we refactor all states

Reply:
True, would be a bit tedious to rename all those booleans, `is<BooleanVar>` it's is

Reply:
I created an issue to revisit this in the future #844